src/node/snapshot_serdes.h

```diff
@@ -58,6 +58,9 @@ namespace ccf
       throw std::logic_error("No receipt included in snapshot");
     }
 
+    LOG_INFO_FMT("Deserialising snapshot receipt (size: {}).", receipt_size);
+    LOG_INFO_FMT("{}", ds::to_hex(receipt_data, receipt_data + receipt_size));
```

Comment:
When a prefix of the file is 0s, I assume `tx_hdr.size` is also 0, and so `receipt_size` is nearly (shrunk by exactly `sizeof(ccf::kv::SerialisedEntryHeader)`) `size`? That's potentially a lot of 0s!

I think that `tx_hdr.size` should also throw an error, _and_ we should cap the number of bytes we're willing to dump as hex here.